flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskManagerJobMetricGroup.java

```diff
@@ -61,9 +58,7 @@ public TaskManagerJobMetricGroup(
 			JobID jobId,
 			@Nullable String jobName) {
 
-		super(registry, jobId, jobName, scopeFormat.formatScope(parent, jobId, jobName));
-
-		this.parent = checkNotNull(parent);
+		super(registry, checkNotNull(parent), jobId, jobName, scopeFormat.formatScope(parent, jobId, jobName));
```

Comment:
But then don't do it here but in the `formatScope` method. Usually the method should enforce that it's preconditions are kept.